internal/cmd/skupper/listener/nonkube/listener_status.go

```diff
@@ -2,28 +2,133 @@ package nonkube
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.listenerHandler = fs.NewListenerHandler(cmd.namespace)
+}
+
+func (cmd *CmdListenerStatus) ValidateInput(args []string) []error {
+	var validationErrors []error
+	opts := fs.GetOptions{RuntimeFirst: true}
+	resourceStringValidator := validator.NewResourceStringValidator()
+	outputTypeValidator := validator.NewOptionValidator(common.OutputTypes)
+
+	// Validate arguments name if specified
+	if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if len(args) == 1 {
+		if args[0] == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+		} else {
+			ok, err := resourceStringValidator.Evaluate(args[0])
+			if !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+			} else {
+				cmd.listenerName = args[0]
+			}
+		}
+	}
+	// Validate that there is a listener with this name in the namespace
+	if cmd.listenerName != "" {
+		listener, err := cmd.listenerHandler.Get(cmd.listenerName, opts)
+		if listener == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener %s does not exist in namespace %s", cmd.listenerName, cmd.namespace))
+		}
+	}
+
+	if cmd.Flags.Output != "" {
+		ok, err := outputTypeValidator.Evaluate(cmd.Flags.Output)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
+		} else {
+			cmd.output = cmd.Flags.Output
+		}
+	}
+
+	return validationErrors
 }
 
-func (cmd *CmdListenerStatus) ValidateInput(args []string) []error { return nil }
-func (cmd *CmdListenerStatus) InputToOptions()                     {}
 func (cmd *CmdListenerStatus) Run() error {
-	return fmt.Errorf("command not supported by the selected platform")
+	opts := fs.GetOptions{RuntimeFirst: true}
+	if cmd.listenerName == "" {
+		listeners, err := cmd.listenerHandler.List()
+		if listeners == nil || err != nil {
+			fmt.Println("no listeners found:")
+			return err
+		}
+		if cmd.output != "" {
+			for _, listener := range listeners {
+				encodedOutput, err := utils.Encode(cmd.output, listener)
+				if err != nil {
+					return err
+				}
+				fmt.Println(encodedOutput)
+			}
+		} else {
+			tw := tabwriter.NewWriter(os.Stdout, 8, 8, 1, '\t', tabwriter.TabIndent)
+			_, _ = fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s\t%s",
+				"NAME", "STATUS", "ROUTING-KEY", "HOST", "PORT"))
+			for _, listener := range listeners {
+				status := "Not Ready"
+				if listener.IsConfigured() {
+					status = "Ok"
+				}
+				fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s\t%d",
+					listener.Name, status, listener.Spec.RoutingKey, listener.Spec.Host, listener.Spec.Port))
+			}
+			_ = tw.Flush()
+		}
+	} else {
+		listener, err := cmd.listenerHandler.Get(cmd.listenerName, opts)
```

Comment:
When I am running `skupper --platform podman listener status backend` against a non-initilized site,
I am getting a warning message twice, saying "**Site not initialized yet**":

```
$ skupper --platform podman listener status backend
Site not initialized yet
Site not initialized yet
Name:		backend
Status:		Not Ready
Routing key:	backend-8080
Host:		127.0.0.1
Port:		8080
```

Reply:
good catch, it came in during the change to remove the update command.  Now fixed.